acnportal/acnsim/network/charging_network.py

```diff
@@ -89,20 +99,34 @@ def register_evse(self, evse, voltage, phase_angle):
         self._EVSEs[evse.station_id] = evse
         self._voltages[evse.station_id] = voltage
         self._phase_angles[evse.station_id] = phase_angle
+        # Update the numpy vector of angles by reconstructing it.
+        self.angles_vector = np.array([angle for _, angle in sorted(self._phase_angles.items())])
 
     def add_constraint(self, current, limit, name=None):
-        """ Add constraint to the network's constraint set.
-
-        Wraps ConstraintSet add_constraint method, see its description for more info.
-        """
-        self.constraint_set.add_constraint(current, limit, name)
+        """ Add an additional constraint to the constraint DataFrame.
 
-    def is_feasible(self, load_currents, t=0, linear=False):
-        """ Return if a set of current magnitudes for each load are feasible at the given time, t.
+        Args:
+            current (Current): Aggregate current which is constrained. See Current for more info.
+            limit (float): Upper limit on the aggregate current.
+            name (str): Name of this constraint.
 
-        Wraps ConstraintSet is_feasible method, see its description for more info.
+        Returns:
+            None
         """
-        return self.constraint_set.is_feasible(load_currents, self._phase_angles, t, linear)
+        if name is None:
+            name = '_const_{0}'.format(len(self.constraints.index))
+        for station_id in current.index:
+            if station_id not in self._EVSEs:
+                raise KeyError('Station {0} not found. Register station {0} to add constraint {1} to network.'.format(station_id, name))
+        current.name = name
+        self.magnitudes[name] = limit
+        # Update numpy vector of magnitudes (limits on aggregate currents) by reconstructing it.
+        self.magnitude_vector = self.magnitudes.sort_index().to_numpy()
+        self.constraints = self.constraints.append(current).sort_index(axis=1).fillna(0)
```

Comment:
Build this dataframe in a new method and use it in `add_constraint` instead of storing it in the network.